backend/src/main/java/com/google/blackswan/mock/DummyAlertGenerator.java

```diff
@@ -22,13 +22,13 @@
   private static final int SET_ALERT_GROUP_SIZE = 1;
 
   private List<Alert> alerts;
-  private AnomalyGenerator anomalyGenerator;
 
-  public DummyAlertGenerator() {
-    anomalyGenerator = new DummyAnomalyGenerator();
+  public DummyAlertGenerator(AnomalyGenerator anomalyGenerator) {
```

Comment:
Btw just for your reference, in Google Java, we heavily use ["Dependency Injection"](https://en.wikipedia.org/wiki/Dependency_injection#:~:text=In%20software%20engineering%2C%20dependency%20injection,object%20is%20called%20a%20service.), using the [Guice framework](https://github.com/google/guice/wiki/GettingStarted).

Feel free to read up on it on your free time, but given that it's quite complicated and take some ramp-up, I probably would not recommend using this framework for MVP, and depending on how complicated things are, maybe not even for the post-MVP. But I still think it's good to take a look briefly, since dependency injection relates to classes taking in other classes as dependencies and inputs.

Reply:
An example of constructor. with dependeny injection using Guice: https://docs.google.com/document/d/1SHC0dkFYnB2NN0tzZAlV5O1TMj955kdYS2OFuQCmqcs/edit#bookmark=id.x1ctn7me7brt (note the `@Inject` keyword).

Reply:
thanks for pointing this out. i will read more about it and hopefully we can incorporate it in for post-MVP.